lifetimes/generate_data.py

```diff
@@ -56,11 +56,71 @@ def beta_geometric_nbd_model(T, r, alpha, a, b, size=1):
             alive = np.random.random() > p
 
         times = np.array(times).cumsum()
-        df.iloc[i] = len(times), np.max(times if times.shape[0] > 0 else 0), T[i], l, p, alive, i
+        df.iloc[i] = len(set(int(t) for t in times)), np.max(times if times.shape[0] > 0 else 0), T[i], l, p, alive, i
 
     return df.set_index('customer_id')
 
 
+def beta_geometric_nbd_model_transactional_data(T, r, alpha, a, b, start_date='2019-1-1', size=1):
+    """
+    Generate artificial transactional data according to the BG/NBD model.
+
+    See [1] for model details
+
+    Parameters
+    ----------
+    T: int, float or array_like
+        The length of time observing new customers.
+    r, alpha, a, b: float
+        Parameters in the model. See [1]_
+    start_date: date_like
+        The date observation starts
+    size: int, optional
+        The number of customers to generate
+
+    Returns
+    -------
+    DataFrame
+        The following columns:
+        'customer_id', 'date'
+
+    References
+    ----------
+    .. [1]: '"Counting Your Customers" the Easy Way: An Alternative to the Pareto/NBD Model'
+       (http://brucehardie.com/papers/bgnbd_2004-04-20.pdf)
+
+    """
+    if type(T) in [float, int]:
+        T = T * np.ones(size)
+    else:
+        T = np.asarray(T)
+
+    start_date = pd.to_datetime(start_date)
+    probability_of_post_purchase_death = stats.beta.rvs(a, b, size=size)
+    lambda_ = stats.gamma.rvs(r, scale=1. / alpha, size=size)
+
+    columns = ['customer_id', 'date']
+    df = pd.DataFrame(columns=columns)
+
+    for i in range(size):
+        p = probability_of_post_purchase_death[i]
+        l = lambda_[i]
+        age = T[i]
+
+        purchases = [[i, start_date - pd.DateOffset(days=1)]]
```

Comment:
mm but what if the user is using a different frequency, like hours or weeks?

Reply:
Thanks. I haven't considered this before. I've added `freq` argument in this function that could be used for different frequency and I've tested the use of it.